git/gogit/client.go

```diff
@@ -343,7 +343,7 @@ func (g *Client) providerAuth(ctx context.Context) error {
 			}
 			switch g.authOpts.ProviderOpts.Name {
 			case git.ProviderAzure:
-				g.authOpts.ProviderOpts.AzureOpts = append(g.authOpts.ProviderOpts.AzureOpts, azure.WithProxyURL(proxyURL))
+				g.authOpts.ProviderOpts.AuthOpts = append(g.authOpts.ProviderOpts.AuthOpts, auth.WithProxyURL(*proxyURL))
```

Comment:
Why is it `*proxyURL` and not `proxyURL` like for github?

Reply:
Because this way when you call WithProxyURL() you can't pass nil, this function receives a copy of the object, with the intention to say: "hey, a proxy URL is not optional in this function, you have have a real URL object here to call this function"